addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -600,3 +612,30 @@ def _format_date_display(self, state, delay):
     @api.model
     def _has_attachments(self, data):
         return data['attachment_ids'] or any(self._has_attachments(component) for component in data.get('components', []))
+
+    def _merge_components(self, component_1, component_2):
+        qty = component_2['quantity']
+        component_1["quantity"] = component_1["quantity"] + qty
+        component_1["base_bom_line_qty"] = component_1["quantity"] + qty
```

Comment:
This won't work if both lines have different units of measure.

Reply:
Mathieu just updated the fact the uoms need to be the same